### PR TITLE
Merge changes from upstream

### PR DESCRIPTION
**Pull request by krnowak**
Looks like Stefan's work was merged into upstream finally.

### CODE REVIEW

File: gexpect_test.go [hunk 1]
```diff
@@ -188,6 +188,31 @@ func TestRegexFind(t *testing.T) {
...
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	s, err := child.ReadLine()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	if s != "foo\r" {
```
Comment (alban) [inline gexpect_test.go:204]: Why does it expect "\r" if the command prints "\n"?

File: gexpect.go [hunk 2]
```diff
@@ -378,27 +378,26 @@ func (expect *ExpectSubprocess) Interact() {
...
 	chunk := make([]byte, 255)
 
 	for {
-
 		n, err := expect.buf.Read(chunk)
 
-		if err != nil {
-			return join, err
-		}
-
```
Comment (alban) [inline gexpect.go:391]: Shouldn't you do the same kind of changes in `Expect()`, line 320?

File: gexpect_test.go (hunk 1 above)
Comment (krnowak) [inline gexpect_test.go:204]: My earlier experiments shown that lines read from pty always end in \r\n.

Comment (krnowak) [inline gexpect_test.go:204]: The comment above was a bit too short - they end in \r\n and ReadLine reads pty until it finds \n, so it returns a line ending with \r.

File: gexpect.go (hunk 2 above)
Comment (krnowak) [inline gexpect.go:391]: Dunno, it is not a change I have made, just some fix from upstream. But probably yes, would need a test for it to know.

**Comment by alban** [comment]
Do you have an opinion how to best keep this in sync with [upstream](https://github.com/ThomasRooney/gexpect)? It will help Debian packaging if rkt uses a release of gexpect instead of a commit hash. But how do we manage versions numbers from ThomasRooney/gexpect and from coreos/gexpect? I don't know if Debian has packaged ThomasRooney/gexpect already.

/cc @steveeJ @onlyjob

Let's see if there is any regression in rkt: https://github.com/coreos/rkt/pull/2346

**Comment by alban** [comment]
I see that the name of the Debian package is namespaced with "github-coreos":
https://packages.debian.org/sid/golang-github-coreos-gexpect-dev

**Comment by krnowak** [comment]
Upstream is not very active, so sometimes we have to wait really long until the PR gets addressed/merged, so for time being I would prefer to stick to our fork. Also, upstream didn't do any releases so far.

**Comment by krnowak** [comment]
Also, no regressions - https://semaphoreci.com/coreos/rkt/branches/pull-request-2346/builds/1

**Comment by alban** [comment]
:+1: 

LGTM

**Comment by alban** [comment]
https://github.com/coreos/gexpect/releases/tag/v0.1.0

**Comment by onlyjob** [comment]
Thanks for keeping me in loop guys. :)

**Pull request merged**